### ISSUE TITLE
Could not find a version to edit for app 'AppName' for 'IOS'

### ISSUE THREAD

**Issue by maxx-cyber** [fastlane/fastlane#17635]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

If there is **Ready for Sale** version in App Store Connect only and there isn't **Prepare for Submission** version, **deliver** command creates **Prepare for Submission** version successfully but anyway failed with message **Could not find a version to edit for app 'AppName' for 'IOS'** without delivering anything.

##### Command executed

fastlane upload_screenshots --env "upload_screenshots"

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>[16:41:25]: Loading from './fastlane/.env.upload_screenshots'
[16:41:25]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[16:41:26]: ---------------------------
[16:41:26]: --- Step: opt_out_usage ---
[16:41:26]: ---------------------------
[16:41:26]: Disabled upload of used actions
[16:41:26]: ----------------------------------------
[16:41:26]: --- Step: Verifying fastlane version ---
[16:41:26]: ----------------------------------------
[16:41:26]: Your fastlane version 2.167.0 matches the minimum requirement of 2.150.1  ✅
[16:41:26]: ------------------------------
[16:41:26]: --- Step: default_platform ---
[16:41:26]: ------------------------------
[16:41:26]: Driving the lane 'ios upload_screenshots' 🚀
[16:41:26]: ---------------------
[16:41:26]: --- Step: deliver ---
[16:41:26]: ---------------------
[16:41:26]: Login to App Store Connect (MODIFIED)
[16:41:30]: Login successful

+--------------------------------------+-----------------------------------------------------------+
|                                     deliver 2.167.0 Summary                                      |
+--------------------------------------+-----------------------------------------------------------+
| force                                | true                                                      |
| skip_metadata                        | true                                                      |
| skip_binary_upload                   | true                                                      |
| overwrite_screenshots                | true                                                      |
| run_precheck_before_submit           | false                                                     |
| app_version                          | 5.4                                                       |
| screenshots_path                     | ./fastlane/screenshots                                    |
| metadata_path                        | ./fastlane/metadata                                       |
| username                             | MODIFIED                                   |
| app_identifier                       | MODIFIED                                 |
| platform                             | ios                                                       |
| edit_live                            | false                                                     |
| use_live_version                     | false                                                     |
| skip_screenshots                     | false                                                     |
| skip_app_version_update              | false                                                     |
| submit_for_review                    | false                                                     |
| reject_if_possible                   | false                                                     |
| phased_release                       | false                                                     |
| reset_ratings                        | false                                                     |
| team_name                            | MODIFIED                                            |
| dev_portal_team_id                   | MODIFIED                                                |
| precheck_default_rule_level          | warn                                                      |
| ignore_language_directory_validation | false                                                     |
| precheck_include_in_app_purchases    | true                                                      |
+--------------------------------------+-----------------------------------------------------------+

[16:41:36]: Making sure the latest version on App Store Connect matches '5.4'...
[16:41:39]: Successfully set the version to '5.4'
[16:41:43]: ------------------------------
[16:41:43]: --- Step: slack_if_enabled ---
[16:41:43]: ------------------------------
[16:41:43]: Parameter enable slack: true
Slack message: Could not find a version to edit for app 'MODIFIED' for 'IOS'
[16:41:43]: Successfully sent Slack notification
+------------------+------------------------+
|               Lane Context                |
+------------------+------------------------+
| ENVIRONMENT      | upload_screenshots     |
| DEFAULT_PLATFORM | ios                    |
| PLATFORM_NAME    | ios                    |
| LANE_NAME        | ios upload_screenshots |
+------------------+------------------------+
[16:41:43]: Could not find a version to edit for app 'MODIFIED' for 'IOS'

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | opt_out_usage              | 0           |
| 2    | Verifying fastlane version | 0           |
| 3    | default_platform           | 0           |
| 💥   | deliver                    | 16          |
| 5    | slack_if_enabled           | 0           |
+------+----------------------------+-------------+

[16:41:44]: fastlane finished with errors

[!] Could not find a version to edit for app 'MODIFIED' for 'IOS'
</pre>
</details>

### Environment

<details>
  <pre><details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15.7                                                        |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.15.7 (19H15)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode_11.7.app/Contents/Developer/               |
| Xcode Version               | 11.7                                                           |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/KrauseFx/fastlane/tree/master/docs
# All available actions: https://github.com/KrauseFx/fastlane/blob/master/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# By default, fastlane will send which actions are used
# No personal data is shared, more information on https://github.com/fastlane/enhancer
# Uncomment the following line to opt out
opt_out_usage

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.150.1"

default_platform :ios


platform :ios do
  before_all do
    # increment_build_number

    # xctool # run the tests of your app
  end
  
  desc "Upload screenshots to App Store Connect"
  lane :upload_screenshots do
    deliver(
      force: true,
      skip_metadata: true,
      skip_binary_upload: true,
      overwrite_screenshots: true,      
      run_precheck_before_submit: false,
      app_version: "5.4"
    )
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful
  end

  error do |lane, exception|
    slack_if_enabled(
      message: exception.message,
      success: false
    )
  end
end


# More information about multiple platforms in fastlane: 
# https://github.com/KrauseFx/fastlane/blob/master/docs/Platforms.md

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier MODIFIED # The bundle identifier of your app
apple_id MODIFIED # Your Apple email address

# You can uncomment any of the lines below and add your own
# team selection in case you're in multiple teams
# team_name "Felix Krause"
team_id MODIFIED

# To select a team for iTunes Connect use
itc_team_name MODIFIED
# itc_team_id "18742801"

# you can even provide different app identifiers, Apple IDs and team names per lane:
# https://github.com/KrauseFx/fastlane/blob/master/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.167.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.3.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 2.3.6        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.3.0        |
| xcodeproj                 | 1.18.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 4.0.6        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.76.0       |
| faraday                   | 1.0.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| json                      | 2.1.0        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.6        |
| bundler                   | 1.17.2       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.2.2        |
| uber                      | 0.1.0        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.15.0       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.1        |
| googleauth                | 0.13.1       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.3        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-storage      | 1.28.0       |
| emoji_regex               | 3.0.0        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.364.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.2        |
| aws-sdk-core              | 3.105.0      |
| aws-sdk-kms               | 1.37.0       |
| aws-sdk-s3                | 1.79.1       |
| forwardable               | 1.2.0        |
| logger                    | 1.3.0        |
| stringio                  | 0.0.2        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.1.0        |
| strscan                   | 1.0.0        |
| date                      | 2.0.0        |
| fileutils                 | 1.1.0        |
| etc                       | 1.0.1        |
| io-console                | 0.4.7        |
| zlib                      | 1.0.0        |
| libxml-ruby               | 3.1.0        |
| rexml                     | 3.1.9        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.4.2        |
</details>

*generated on:* **2020-11-19**
</details></pre>
</details>

**Comment by maxx-cyber** [comment]
As a workaround I've separated deliver command in a lane on two deliver commands, first one for create/update version, next one for uploading screenshots. It works but it leads to double login to App Store Connect.

**Comment by joshdholtz** [comment]
@maxx-cyber Heyyy 👋 This is really strange looking 🤔 

1. What is the version number that is “Ready for Sale”?
2. The version number is should be creating is `5.4`, correct?
3. Are there other platforms for this app besides `ios`?

Thanks!

**Comment by maxx-cyber** [comment]
1. 5.3
2. Yes
3. No, we use Fastlane for iOS apps only, but we have a lot of iOS apps and this issue is 100% reproducible

**Comment by maxx-cyber** [comment]
@joshdholtz also seems this issue is reproducible for uploading metadata as well, but it works correctly if uploading ipa only without screenshots and metadata.

**Comment by joshdholtz** [comment]
@maxx-cyber I actually just got a personal app approved so I will run your exact setup again my app to see if I can reproduce 🤞  I’ll try and have an answer to you early next week if that’s okay!

**Comment by maxx-cyber** [comment]
ok
BTW uploading ipa without screenshots and metadata works because uploading ipa doesn't relate to app version during uploading

**Comment by julianD77** [comment]
We have also experienced this new issue today, however our `deliver` command is already set to skip the metadata and does not upload screenshots:

Error shown from Fastlane output on CI

>  Could not find a version to edit for app 'xxxxx' for 'IOS'

Delivery Summary

> skip_metadata                        | true  
> skip_screenshots                   | false  

Fastfile snippet
>
`deliver(
      force: true, 
      app_identifier: "xxxxxx", 
      skip_metadata: true
    )`

UPDATE: on a retry of the same build/fastlane setup, it worked. Hmmm

**Comment by maxx-cyber** [comment]
Hi @joshdholtz 
Any updates here?

**Comment by joshdholtz** [comment]
Hey, sorry! Me and my wife just had a baby so a little backlogged on things 😔 Will make sure I look into this as soon as I can for you!

**Comment by joshdholtz** [comment]
@maxx-cyber Hey, sorry! Me and my wife just had a baby so a little backlogged on things 😔 Will make sure I look into this as soon as I can for you!

**Comment by CovertLeopard** [comment]
@maxx-cyber @joshdholtz  I'm not sure if this helps in any way, but we ran into this issue on 12/8.  Our support team didn't tell anybody until just today, which is how I found this ticket.  

Apparently, something happened during a build or prior to the build (still not sure there), where a version was created on the Apple Dev account that was the same version we were trying to push using Fastlane, but it wasn't editable (had already been released).  We bumped the build number by 1 in our build system database so that it would have to create a new version in the Apple Dev account.  Doing this allowed Fastlane to operate as expected and it built the incremented version and submitted it just fine.  As a note, we are not currently on the lastest Fastlane (I know...), but this worked to get around this error on Fastlane 2.161.0.
